src/find-controllers.ts

```diff
@@ -11,34 +12,56 @@ export type FindControllersResult = Array<IStateAndTarget>
  *
  * @param pattern Glob pattern
  * @param opts Glob options
+ * @param esmModules Set to true to return controller resolution results as a Promise, to allow ESM dynamic imports rather than CJS "require"
  */
 export function findControllers(
   pattern: string,
   opts?: glob.Options,
-): FindControllersResult {
+  esmModules?: true,
```

Comment:
1. The option should be named `esModules` (consistent with Awilix)
2. How about defining a `FindControllersOptions` that is a union of `{ esModules?: boolean } | glob.Options` instead? 

Also, the return type should be conditional, similar to Awilix, to avoid breaking the types.